### PR TITLE
Vagrantfile improvements

### PR DESCRIPTION
**Pull request by fatz**
- added version contraint for vagrant. Stops the run before suspicious bash provider errors
- some ruby based cleanups. Empty lines, method names, string interpolation, regex call...

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -174,28 +170,23 @@ Vagrant.configure(2) do |config|
   config.hostmanager.ignore_private_ip = false
 
   # configure vagrant-vbguest plugin
-  if Vagrant.has_plugin?('vagrant-vbguest')
```
Comment (karlkfi) [inline Vagrantfile]: prefer this not be inline. gonna have to add another plugin check soon, want them to be easy to spot.

**Comment by karlkfi** [comment]
just one nitpick. otherwise, looks good. thanks!

File: Vagrantfile (hunk 1 above)
Comment (fatz) [inline Vagrantfile]: hmm ok... in this case what do you think about these options

1. `&&` 
  ```ruby
  # configure vagrant-vbguest plugin
  Vagrant.has_plugin?('vagrant-vbguest') && config.vbguest.auto_update = true
  ```

2. verbose comments ( well actually I made this 'cause of this cop https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/if_unless_modifier.rb#L17 )

  ```ruby
  # Vagrant Plugin Configuration: vagrant-vbguest
    if Vagrant.has_plugin?('vagrant-vbguest')
      # enable auto update guest additions
      config.vbguest.auto_update = true
    end
  ```
  its just keeping the rule not doing single line blocks.

there is also an third option deactivating the cop. But it adds a useless line for everybody not using rubocop 😄 

i also can change it back to what you did so you can accept the rest.

just let me know

**Comment by karlkfi** [comment]
silly rubocop, whining about trivial things...

It's nice and readable the way it is. I want it to be obvious so people can disable it easily if they want, without having to add yet another environment variable. 

I'm ok with having line comments tho. If you prefer that.

**Comment by fatz** [comment]
ok updated it...

and indeed its a bit trivial... however i personally like tools like rubocop to keep things clean

**Comment by karlkfi** [comment]
some recent PRs were just merged. can you rebase?

**Comment by fatz** [comment]
ok did the rebase but please double check... had conflicts with `lib/vagrant-dcos/provisioner.rb`. Guess i resolved them but please review it.

**Comment by karlkfi** [comment]
oops. guess i should have merged this first. sorry. needs another rebase.

**Comment by fatz** [comment]
no worries.. on it

**Comment by karlkfi** [comment]
great! Thanks!

**Pull request merged**